File: multi-module-sample-java/module-api/src/main/java/com/sample/app/controller/ServerController.java
package com.sample.app.controller;

import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@Slf4j
@RestController
@RequestMapping("/server")
public class ServerController {

    @Value("${core.info}")
    private String coreInfo;

    @GetMapping("/core")
    public ResponseEntity<String> coreInfo() {
        return ResponseEntity.ok().body(coreInfo);
    }
}

File: multi-module-sample-kotlin/module-api/src/main/kotlin/com/sample/app/ServerController.kt
package com.sample.app

import com.sample.app.config.ServerConfig
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController
import javax.validation.Valid

@RestController
@RequestMapping(value = ["/server"])
class ServerController(val serverConfig: ServerConfig) {


    @RequestMapping(value = ["/core/version"])
    fun getCoreInfo(): ResponseEntity<String> = ResponseEntity.ok().body(serverConfig.getCoreVersion())
}